### PR TITLE
Adding a detailed overview diagram to appendix

### PR DESCRIPTION
**Pull request by shigeya**
**Experimental**
Keeping old figure for review.

Related to #453.  Trying to include all of the relationships of the entities described in 1.3. Architecture Overview.

This diagram does not display DID methods, DID resolvers, and DID URL dereferences in this picture since their position in this level of abstraction is unclear.

***
<a href="https://pr-preview.s3.amazonaws.com/shigeya/did-core/pull/600.html" title="Last updated on Feb 10, 2021, 4:34 AM UTC (36b8395)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/600/59687a4...shigeya:36b8395.html" title="Last updated on Feb 10, 2021, 4:34 AM UTC (36b8395)">Diff</a>

### CODE REVIEW

**Comment by shigeya** [comment]
Note that we need consistency of notation/expression/sizes/whatever among the figures to be consistent.
I believe I can update figures as part of editorial fixes. So, I let the color black and white for now.

**Comment by shigeya** [comment]
> This diagram does not display DID methods, DID resolvers, and DID URL dereferences in this picture since their position in this level of abstraction is unclear.

I tried to place them into the figure. Texts are smaller than I want, but now every component described in the section is in the diagram.

**Comment by shigeya** [comment]
The preview link above doesn't work. For the review, I added the following PNG here.

did_architecture_overview-2

**Comment by shigeya** [comment]
Minor fix

did_architecture_overview-2

**Comment by shigeya** [comment]
Minor update to add self-reference dereference to DID Document.
did_architecture_overview-2

**Comment by shigeya** [comment]
PR includes the SVG file for the above figure. This time, I also committed the corresponding PNG file too.

**Review by OR13 — APPROVED** [review]
This diagrams is one of the greatest contributions I am seen... massively helpful.

**Comment by shigeya** [comment]
Clean-up of the PR. Also fixed the text `DID document-local` to `DID document-relative.`
did_architecture_overview-2

**Comment by shigeya** [comment]
Two issues on the figure
1. It's correctly drawn with Safari but not with Firefox(85.0) and Chrome(88), as attached image (properties inside the DID document box): ScreenShot 2021-02-08 17 18 45
2. Probably need to draw a relationship between DID resolver and DID dereferencer.

**Comment by shigeya** [comment]
Fixed the compatibility issue mentioned above.

File: diagrams/did_architecture_overview-2.svg [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+          </text>
+        </g>
+      </g>
+      <g id="Line_15">
+        <path d="M 223.98106 152.54618 C 215.57116 158.46523 202.18972 170.02648 189.92127 189.92127 C 170.11222 222.04404 162.51969 260.7874 162.51969 260.7874" marker-start="url(#FilledArrow_Marker)" stroke="black" stroke-linecap="round" stroke-linejoin="round" stroke-width="1"/>
+      </g>
+      <g id="Graphic_64">
+        <text transform="translate(175.07875 305.47245)" fill="black">
+          <tspan font-family="Helvetica Neue" font-size="12" font-weight="400" fill="black" x="7531753e-19" y="11">fragment refers, and</tspan>
+          <tspan font-family="Helvetica Neue" font-size="12" font-style="italic" font-weight="700" fill="black" x="19.332" y="26.336">dereferences</tspan>
```
Comment (TallTed) [inline diagrams/did_architecture_overview-2.svg:163]:
```suggestion
          <tspan font-family="Helvetica Neue" font-size="12" font-style="italic" font-weight="700" fill="black" x="19.332" y="26.336">dereferences,</tspan>
```

Comment (shigeya) [reply diagrams/did_architecture_overview-2.svg]: Thanks for the suggestion. fixed.

**Review by msporny — APPROVED** [review]
While I agree that the diagram is accurate -- it exposes the reader to a large number of details up front and makes the DID ecosystem look horribly complicated.

I would prefer to have the diagram in an appendix -- we went to great lengths in the Verifiable Credentials specification to simplify the diagram in the overview because we knew people would cut/paste it into their presentations. That bet paid off in a big way.

While this diagram is accurate, I'm very concerned about people copying it into their presentations -- it'll make the specification and technology come off as a complex, academic, and inaccessible.

I'm approving the diagram because I do think it's very useful -- but in an appendix, after people have read through the entire spec (or the more complicated portions of it). I think we need a summary diagram in the overview section... and the overview section needs to be simplified.

**Comment by shigeya** [comment]
@msporny I understand your concern and I fully agree with you. Let me try to create a simpler diagram based on this in 13 hours (It's about 11 am in AoE clock: https://time.is/Anywhere_on_Earth).

**Comment by shigeya** [comment]
A brief version of the above is created and sent as another PR #612. I will update this PR to relocate the diagram to the appendix section.

**Comment by shigeya** [comment]
Relocated to the appendix, prior to FAQ. Attached the image for the convenience of the review.

To make use of the rich real estate of the appendix section, adjusted the diagram to be narrower, taller. Adjusted to use larger font size and little extra spaces between texts, boxes, and lines.

Plus, some minor clean-ups including "did:" to be text, not in a box.
did_detailed_architecture_overview

**Comment by msporny** [comment]
Editorial, multiple reviews, changes requested and made, no objections, merging.

**Pull request merged**